src/ServicePulse.Host/app/js/services/services.monitoring.js

```diff
@@ -43,19 +43,24 @@
         }
 
         function createEndpointDetailsSource(endpointName, sourceIndex, historyPeriod) {
-            return Rx.Observable.interval(5000).startWith(0)
+            return Rx.Observable.interval(calculateIntervalInMiliseconds(historyPeriod)).startWith(0)
                 .flatMap(function (i) {
                     return Rx.Observable.fromPromise(loadEndpointDetailsFromMonitoringService(endpointName, sourceIndex, historyPeriod));
                 });
         }
 
+        function calculateIntervalInMiliseconds(period) {
+            return (period * 1000 * 60) / 20;
```

Comment:
Every 1 minute? Shouldn't we refresh more frequently than the window?

Reply:
☝️  Assuming the user has selected "The last 1 minute" as their window

Reply:
if period is 1 than
1 * 1000 * 60 = 60000
60000 / 20 = 3000 ms = 3 sec
So for period 1 minute we refresh every 3 seconds, which is a length of the interval. 

For 5 minutes period it will be 15 seconds (also the length of the interval). At the moment we query every 5 seconds. But it doesn't make sense to query more frequently than interval length.

Reply:
Oops, missed the `/ 20` part!

Reply:
What if we added the refresh interval to `historyPeriods` as third property? We could hardcode it there and not require `calculateIntervalInMiliseconds`.

Reply:
I'm ok with using a heuristic calculation.